deeprank/learn/DataSet.py

```diff
@@ -283,9 +283,10 @@ def process_dataset(self):
 
         # get the pairing
         self.get_pairing_feature()
-
-        # get grid shape
-        self.get_grid_shape()
+        
+        if self.grid_shape is None: 
```

Comment:
The `if self.grid_shape is None` condition has been considered in the function `get_grid_shape`, see the line 824 of the same file.

Reply:
Indeed but then the following `elif` condition is not correctly placed (line 839). It is more logical to remove the `if self.grid_shape is None` from that function and to enter it only if it does not fit exist.
This same issue is pointed out in #242

Reply:
Thanks, I'll close this PR then.